### DISCUSSION TITLE
Brightnessctl is not working without root permissions.

### DISCUSSION THREAD

**Discussion by OmShivamR24** [i3/i3#4763] [Q&A] [answered]
Hello, when I am trying to control my screen brightness through brightnessctl in I3 config, it is not working. Manually typing out sudo brightnessctl set +5% and such works but not when I bind it to a key.
My guess is that it works only when prepended with sudo.

So my question is how to access brightnessctl w/o root permissions.

I have tried copying /usr/bin/brightnessctl to home/.local/bin but it still asks for. Root permission.

Please advice on how to proceed.
Thank you.

**Comment by Airblader** [comment, marked as answer]
You can allow running the command with passwordless sudo — search for that term to find instructions on how to do that and a discussion around security implications. As such, this question has nothing to do with i3.

**Comment by orestisfl** [comment]
See the readme of brightnessctl for instructions on how to use it without sudo.

Or maybe see https://github.com/haikarainen/light

**Reply by junishigohoka** [reply]
The link is not working...

**Reply by orestisfl** [reply]
https://gitlab.com/dpeukert/light seems to be what the Arch package uses

**Comment by Eliot-Roxbergh** [comment]
The following fixed it for me; use the provided udev rules [1] and add current user to video group i.e. run `usermod -a -G video ${USER}` and logout and in again. This was also mentioned here https://github.com/Hummer12007/brightnessctl/issues/63#issue-818994525.

[1] - According to brightnessctl's readme, udev rules is included with the installation. For instance, on debian it is included automatically in [brightness-udev](https://packages.debian.org/bookworm/brightness-udev)

**Reply by rswdch** [reply]
Thank you, this was easy to follow and worked for me on Pop!_os 22.04

**Comment by Walheimat** [comment]
If you don't want to do the things  [brightnessctl](https://github.com/Hummer12007/brightnessctl) suggests, could run `sudo chmod +s /usr/bin/brightnessctl` (or `sudo chmod +s $(which brightnessctl)`).

**Reply by SyntaxOutlaw** [reply]
`sudo chmod +s $(which brightnessctl)`

This was exactly what I was looking for, thank you very much!

**Reply by Akhilbisht798** [reply]
this works thanks.

**Reply by astroajay** [reply]
> If you don't want to do the things [brightnessctl](https://github.com/Hummer12007/brightnessctl) suggests, could run `sudo chmod +s /usr/bin/brightnessctl` (or `sudo chmod +s $(which brightnessctl)`).

Thank you so much! 
This worked perfectly!

**Comment by sean-m-foley** [comment]
It has nothing really to do with i3, honestly...

You have to run any command to change brightness on ANY distro as setuid root or there is no way to access the hardware. You can craft udev rules too, but yeah it's not really an i3 problem.

[light](https://github.com/haikarainen/light/blob/master/90-backlight.rules) uses these rules to change the light command to set group id, and then give the 'video' group (which the user is normally a part of) access to the controls. (They are copied to /usr/lib/udev/rules.d/)

**Comment by ghost** [comment]
```
xrandr --output HDMI-1 --brightness 1
xrandr --output HDMI-1 --brightness 0.5
```

works without root.

to get current output run
`xrandr -q | grep ' connected' | head -n 1 | cut -d ' ' -f1`

**Reply by hcgrove** [reply]
That seems to only understand absolute values, not good for increasing or decreasing brightness, and more importantly:
From  xrandr(1):

> However, this is a software only modification, if your hardware  has  support  to  actually
> change the brightness, you will probably prefer to use xbacklight.

I remember using `xbacklight`, but I somehow concluded that it wasn't as good as something else (nowadays I use `light`, and considers switching to `brightnessctl`, but I might not have gone directly)
